src/pages/index.js

```diff
@@ -0,0 +1,361 @@
+const avatarModalCloseBtn = avatarModal.querySelector(".modal__close-btn");
+const avatarAddButton = document.querySelector(".profile__avatar-edit-button");
+
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+const cardNameInput = cardModal.querySelector("#add-card-name-input");
+
+//Select the modal
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaption = previewModal.querySelector(".modal__caption");
+const previewCloseBtn = previewModal.querySelector(".modal__close-btn");
+
+//To set the template element
+const cardTemplate = document.querySelector("#card-template");
+// console.log(cardTemplate)
+
+//Create a function to call the card list
+const cardsList = document.querySelector(".cards__list");
+
+//Functions
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameEL = cardElement.querySelector(".card__title");
+
+  //Select the image class and it's elements
+  const cardImage = cardElement.querySelector(".card__image");
+
+  //select the card like button element
+  const cardLikedBtn = cardElement.querySelector(".card__like-button");
+  const deleteCardBtn = cardElement.querySelector(".card__delete-button");
+
+  //assign  value to the name
+  cardNameEL.textContent = data.name;
+  //check if the card is liked or not
+
+  //assign  values to the image src and alt
+  cardImage.src = data.link;
+  cardImage.alt = data.name;
+
+  // Safely retrieve and parse liked cards from localStorage
+  let likedCards = {};
+
+  try {
+    likedCards = JSON.parse(localStorage.getItem("likedCards")) || {};
+  } catch (error) {
+    console.error("Error parsing likedCards from localStorage:", error);
+    localStorage.setItem("likedCards", JSON.stringify({})); // Reset storage
+  }
+
+  // Apply liked state
+  if (likedCards[data._id]) {
+    cardLikedBtn.classList.add("card__like-button_liked");
+  }
+
+  //add the event listener
+  cardLikedBtn.addEventListener("click", (evt) => handleLikeCard(evt, data));
+  deleteCardBtn.addEventListener("click", (evt) => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  //add event listener for the card image
+  cardImage.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaption.textContent = data.name;
+  });
+  // console.log("Generated Card:", cardElement); // Debugging
+  return cardElement;
+}
+
+function handleLikeCard(evt, data) {
+  const isLiked = evt.target.classList.contains("card__like-button_liked");
+
+  api
+    .handleLike(data._id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-button_liked");
+
+      let likedCards = JSON.parse(localStorage.getItem("likedCards")) || {};
```

Comment:
* `localStorage` can't be used for **liking**. Everything should be checked according to the task